code-challenges.test.js

```diff
@@ -27,59 +51,138 @@ var num3 = -7
+    expect(arrayCaps(randomNouns2)).toEqual(capResult2)
+  })
+})
+
 var randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
 
 var randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deduction"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango", "Deduction"]
 
-
-
-
-
 // b) Create the function that makes the test pass.
 
+//create a function with an array as an arguement
+//set a variable that will map the arrays
+//the return of the map will have toUpperCase for only the index of 0 and will join back with the vaule with substring.
 
+const arrayCaps = (array) => {
+  let newArrayToCaps = array.map(value => {
+    return value[0].toUpperCase() + value.substring(1)
+  })
+  return newArrayToCaps
+}
 
-
+console.log(arrayCaps(randomNouns1));
+console.log(arrayCaps(randomNouns2));
 
 // --------------------3) Create a function that takes in an array of mixed data types and returns an array with ONLY NUMBERS sorted from least to greatest.
 
 // a) Create a test with expect statements for each of the variables provided.
 
+//this test has helped me to find that when I used sort, it sorts just the positive number value -- not the negative values causing me to refactor my function.
+//
+describe("when numOnly is called", () => {
+  var mixedDataArray1 = [true, 8, "hello", -8, null, 0, 46, 59, 107, "hey!"]
+  var mixedDataArray2 = [3, "yo!", 94, -9, false, 0, 18, "hola!"]
+  let numOnlyResult1 = [-8, 0, 8, 46, 59, 107]
+  let numOnlyResult2 = [-9, 0, 3, 18, 94]
+  it("returns an array of numbers only in ascending order", () => {
+    expect(numOnly(mixedDataArray1)).toEqual(numOnlyResult1)
+  })
+  it("returns an array of numbers only in ascending order", () => {
+    expect(numOnly(mixedDataArray2)).toEqual(numOnlyResult2)
+  })
+})
+
 var mixedDataArray1 = [true, 8, "hello", -8, null, 0, 46, 59, 107, "hey!"]
 // Expected output: [-8, 0, 8, 46, 59, 107]
 
-var mixedDataArray1 = [3, "yo!", 94, -9, false, 0, 18, "hola!"]
+var mixedDataArray2 = [3, "yo!", 94, -9, false, 0, 18, "hola!"]
 // Expected output: [-9, 0, 3, 18, 94]
 
 
+// b) Create the function that makes the test pass.
 
+//make a function that takes an array as an arguement with multiple data types
+//use filter typeof to determine only number data types
 
+//use sort to organize numbers in array from least to greatest --- having a hard time with this sorting properly.  Negative numbers keep appearing with the positive numbers making the test fail.  Have tried to refactor this multiple times to solve this problem and have come up with errors each time.
 
-// b) Create the function that makes the test pass.
-
+//return values as a new array
 
+const numOnly = (array) => {
+  let numArray = array.filter (value => {
+    if (typeof value === 'number')
+      return value
+      })
+    for(let i = 0; i < numArray.length; i++)
```

Comment:
reconsider the placement of this for-loop because it's happening within an iteration (`.filter()`) already. It would read a bit clearer that you first complete the filtering then run the for-loop